### DISCUSSION TITLE
issue creacion de cargo

### DISCUSSION THREAD

**Discussion by JuanSe320** [CriistiianDM/dui-iglesias-software-samuel#52] [Q&A] [answered]
pienso que ya quedo la consulta a la BD que segun el id de la persona le dice los cargos que tiene a su disposicion

**Comment by JuanSe320** [comment]
![image](https://user-images.githubusercontent.com/63017408/179370191-a44ce62d-5dae-4f4e-896e-28a25b933050.png)

**Reply by JuanSe320** [reply]
ese seria la persona con el id 1

**Comment by JuanSe320** [comment]
![image](https://user-images.githubusercontent.com/63017408/179370233-1bb4c119-af8b-44f8-ab28-33f9dc6177b9.png)

**Reply by JuanSe320** [reply]
y aqui esta la persona con el id 2 que tiene dos cargos a su nombre

**Comment by CriistiianDM** [comment, marked as answer]
La consulta esta bien, Pero que no se te olvide que ya con esa consulta sabes los cargos de la persona para no reasignar  cargos que ya tiene

**Reply by CriistiianDM** [reply]
Falta aclarar que la subconsulta sobra, por que hay un campo dentro de la misma tabla que tiene el id de las personas

**Reply by JuanSe320** [reply]
a pero entonces añadirle que los del periodo vigente, pero hay la cosa es que se tendria que saber que periodo esta cursando no?

**Reply by JuanSe320** [reply]
> Falta aclarar que la subconsulta sobra, por que hay un campo dentro de la misma tabla que tiene el id de las personas

aaa si porque cuando loguea tiene el id activo no? entonces seria algo asi

**Reply by JuanSe320** [reply]
![image](https://user-images.githubusercontent.com/63017408/179370489-4c5eca0d-0650-4f9a-9665-2e258acad088.png)

**Reply by JuanSe320** [reply]
porque ya venia el id en locale store si verdad

**Reply by CriistiianDM** [reply]
> a pero entonces añadirle que los del periodo vigente, pero hay la cosa es que se tendría que saber que periodo esta cursando no?

La cosa es que en el campo visual haces reutilizas la consulta que sale en la tabla de administrador con todos los usuarios o déjame que seria mas optimo guardar todos los usuarios el localstore y asi sale mas optimo.

Para iniciar sesión se verifica que los cargos que tengas sean vigente y si no los elimina cosa que no esta hecha por que si elimina tocaria ver que pasa con los demás pero para eso podemos usar el campo de estado pero eso es para despues, ahhhh pero solo al usuario no a los demas...

seleccionas a la persona y le seleccionas el cargo que le quieres asignar y le da guardar y en esa verificación va  hacer un insert a la tabla person_position pero antes de eso puedes validar que  el id del cargo no este en los cargos que ya tiene para poder ingresarlo y si ya lo tiene se le manda un mensage que diga por ejemplo "Ya tiene este cargo"

**Reply by CriistiianDM** [reply]
> porque ya venia el id en locale store si verdad

nope pero ya que veo que lo vamos a estar usando hare añadire al id al localstore

**Reply by CriistiianDM** [reply]
> ![image](https://user-images.githubusercontent.com/63017408/179370489-4c5eca0d-0650-4f9a-9665-2e258acad088.png)

si, asi

**Reply by JuanSe320** [reply]
en la de abajo creo que la mejore para que no muestre cargos repetidos segun el periodo

**Reply by JFOZ1010** [reply]
> La consulta esta bien, Pero que no se te olvide que ya con esa consulta sabes los cargos de la persona para no reasignar cargos que ya tiene

a ver....

**Comment by JuanSe320** [comment]
![image](https://user-images.githubusercontent.com/63017408/179370562-05b18c1e-0cf0-4868-a791-cf59f98bf1df.png)

**Reply by JuanSe320** [reply]
creo que seria algo asi, asi no repetira cargos porque si han pasado dos periodos le saldria tanto los viejos como los nuevos y asi creo que ya quedaria sin ese problema y se  sustituiria la variable del periodo por el que esta en ese momento

**Reply by CriistiianDM** [reply]
creo que el periodo es algo que hay pensarlo bien, por que según entendí,  habrán varios periodos vigentes y cuando se vence el periodo lo que pienso hacer es que esos cargos se pongan en true y ya todas las personas que tengan ese periodo pierden el cargo por igual, la cosa seria que se tendria que reponer de manera manual a cada usuario, por eso te digo que la cuestión de los periodos  no es clara

**Reply by JuanSe320** [reply]
te voy a mandar un audio de eso porque escrito se me hace dificil expresar como creo que funcionan y porque lo coloque en la consulta

**Reply by CriistiianDM** [reply]
Es que pienso desde que inicia sesión y el cargo venció se le quita solo a esa persona, entonces si, validad que el cargo sea vigente cuando le asigna un cargo que ya tiene pero validas que sea vigente, creo que podes usar la funcion que uso en el inicio de sesion que valida si el cargo esta vigente, pero ojo que cuando no esta vigente sale error, por que todavía no se que hacer con eso, si quitar el periodo a esa persona o a todas las personas

**Reply by CriistiianDM** [reply]
> te voy a mandar un audio de eso porque escrito se me hace dificil expresar como creo que funcionan y porque lo coloque en la consulta

si pa casos como este es mejor usar el whatsapp

**Reply by CriistiianDM** [reply]
Si no aqui nos quedamos hablando todo el year

**Comment by JFOZ1010** [comment]
ya.
